pandas/core/indexing.py

```diff
@@ -2166,9 +2168,13 @@ def check_bool_indexer(index: Index, key) -> np.ndarray:
         # key might be object-dtype bool, check_array_indexer needs bool array
         result = np.asarray(result, dtype=bool)
         result = check_array_indexer(index, result)
+    elif not is_array_like(result):
+        # GH 33924
+        # key may contain nan elements, check_array_indexer needs bool array
+        result = pd_array(result, dtype=bool)
```

Comment:
can you make this a bit simpler here, pls do this:

L2166, change that to: `return result.astype(bool)._values`
L21067: elif -> if
rm L2170
rm L2174
rm L2176-2177

L2178: return check_array_indexer(indexer, result)

basically pulling the last 3 clauses together